recognito/src/main/java/com/bitsinharmony/recognito/Recognito.java

```diff
@@ -212,6 +212,55 @@ public VoicePrint createVoicePrint(K userKey, File voiceSampleFile)
 
         return createVoicePrint(userKey, audioSample);
     }
+	
+		/**
+     * Add a voiceprint object. 
+     * <p>
+     * Threading : this method is synchronized to prevent inadvertently erasing an existing user key
+     * </p>
+     * @param userKey the user key associated with this voice print
+     * @param voiceSample the voice print object
+     * @return the voice print itself
+     */
+	
+	public synchronized VoicePrint addVoicePrint(K userKey, VoicePrint voicePrint) {
```

Comment:
Hi, thanks for creating this pr
Better to get feedback late than never, I guess...

I'd think this method is unnecessary: when using the constructor 
Recognito(float sampleRate, Map<K, VoicePrint> voicePrintsByUserKey)
The user already provides all the persisted voicePrints.

Since Recognito creates and stores all instances of VoicePrint, I don't see how an external VoicePrint could be provided.

Also note: having the inner synchronized block seems superfluous because "this" is already locked at method level